src/router/router.js

```diff
@@ -108,8 +111,20 @@ export const navigate = async function () {
     const previousRoute = currentRoute
     const hash = getHash()
     const route = matchHash(hash, this.parent[symbols.routes])
-
+    let beforeHookOutput
     if (route) {
+      if (route.hooks) {
+        if (route.hooks.before) {
+          const args = route.params ? route.params : {}
+          beforeHookOutput = await route.hooks.before(args)
+          if (isString(beforeHookOutput)) {
+            to(beforeHookOutput)
+            return
+          }
+        }
+      }
+      beforeHookOutput = isObject(beforeHookOutput) ? beforeHookOutput : {}
```

Comment:
what does this line do exactly? I'm not sure I understand :)

Reply:
If the beforehook returns something other than an object, such as a function or a number, then map the beforeHookOutput to an empty object so that it won't be a problem with spread operator

Reply:
@michielvandergeest  I have verified all possible types with spread operator like `{...1}`, `{...()=>{}}`, `{...undefined}` and found three is no effect with these values so we can safely ignore this checking. I will remove this statement.